net/apfree-wifidog/files/wifidog.init

```diff
@@ -0,0 +1,248 @@
+	config_get untrusted_maclist 	"$cfg"	"untrusted_maclist"
+	config_get pool_mode		"$cfg"	"pool_mode"			0
+	config_get thread_number	"$cfg"	"thread_number"		20
+	config_get queue_size		"$cfg"	"queue_size"		200
+	config_get wired_passed		"$cfg"	"wired_passed"		1
+	config_get trusted_iplist	"$cfg"	"trusted_iplist"
+	config_get trusted_pan_domains	"$cfg"	"trusted_pan_domains"
+	config_get proxy_port	"$cfg"	"proxy_port"
+	config_get no_auth	"$cfg"	"no_auth"
+	config_get apple_cna	"$cfg"	"bypass_apple_cna"
+	config_get update_domain_interval	"$cfg"	"update_domain_interval"
+	config_get dns_timeout	"$cfg"	"dns_timeout"
+	
+	local set_auth_server_path_login=`[ -n "$auth_server_path_login" ] && echo "    LoginScriptPathFragment $auth_server_path_login"`
+	local set_auth_server_path_portal=`[ -n "$auth_server_path_portal" ] && echo "    PortalScriptPathFragment $auth_server_path_portal"`
+	local set_auth_server_path_msg=`[ -n "$auth_server_path_msg" ] && echo "    MsgScriptPathFragment $auth_server_path_msg"`
+	local set_auth_server_path_ping=`[ -n "$auth_server_path_ping" ] && echo "    PingScriptPathFragment $auth_server_path_ping"`
+	local set_auth_server_path_auth=`[ -n "$auth_server_path_auth" ] && echo "    AuthScriptPathFragment $auth_server_path_auth"` 
+	local set_delta_traffic=`[ -n "$delta_traffic" ] && echo "DeltaTraffic $delta_traffic"` 
+	local set_trusted_maclist=`[ -n "$trusted_maclist" ] && echo "TrustedMACList $trusted_maclist"` 
+	local set_untrusted_maclist=`[ -n "$untrusted_maclist" ] && echo "UntrustedMACList $untrusted_maclist"` 
+	local set_trusted_domains=`[ -n "$trusted_domains" ] && echo "TrustedDomains	$trusted_domains"`
+	local set_trusted_iplist=`[ -n "$trusted_iplist" ] && echo "TrustedIpList	$trusted_iplist"`
+	local set_trusted_pan_domains=`[ -n "$trusted_pan_domains" ] && echo "TrustedPanDomains"	$trusted_pan_domains`
+	local set_proxy_port=`[ -n "$proxy_port" ] && echo "Proxyport"	$proxy_port`
+	local set_no_auth=`[ -n "$no_auth"  ] && echo "NoAuth"  $no_auth`
+	local set_firewall_rule_global=`config_list_foreach "$cfg" "firewall_rule_global" echo_firewall_rule`
+	local set_firewall_rule_validating_users=`config_list_foreach "$cfg" "firewall_rule_validating_users" echo_firewall_rule`
+	local set_firewall_rule_known_users=`config_list_foreach "$cfg" "firewall_rule_known_users" echo_firewall_rule`
+	local set_firewall_rule_auth_is_down=`config_list_foreach "$cfg" "firewall_rule_auth_is_down" echo_firewall_rule`
+	local set_firewall_rule_unknown_users=`config_list_foreach "$cfg" "firewall_rule_unknown_users" echo_firewall_rule`
+	local set_firewall_rule_locked_users=`config_list_foreach "$cfg" "firewall_rule_locked_users" echo_firewall_rule`
+	local set_apple_cna=`[ -n "$apple_cna"  ] && echo "BypassAppleCNA" $apple_cna`
+	local set_update_domain_interval=`[ -n "$update_domain_interval" ] && echo "UpdateDomainInterval" $update_domain_interval`
+	local set_dns_timeout=`[ -n "$dns_timeout" ] && echo "DNSTimeout" $dns_timeout`
```

Comment:
this is extremely complicated, are you sure it needs to be like this? you're invoking dozens of extra shells.

Reply:
all these script was used to make user utilizing apfree-wifidog more easy